app/Console/Migrations/MigrateProjects.php

```diff
@@ -0,0 +1,340 @@
+
+                $projectMapping = new InsightlyMapping(
+                    $integrationId,
+                    $projectId,
+                    ResourceType::Project
+                );
+                $this->insightlyMappingRepository->save($projectMapping);
+            } catch (Exception) {
+                $this->warn($integrationId . ' - Did not find project with id ' . $projectId);
+            }
+        }
+    }
+
+    private function migrateContact(UuidInterface $integrationId, string $contactId): void
+    {
+        $uitIdContact = $this->getContactFromUiTiD($integrationId, $contactId);
+        $email = $uitIdContact?->email;
+        if ($uitIdContact === null || $email === null) {
+            $this->warn($contactId . ' - user not found inside UiTiD');
+            return;
+        }
+
+        $insightlyContacts = $this->insightlyClient->contacts()->findByEmail($email);
+        if ($insightlyContacts->isEmpty()) {
+            $this->warn($contactId . ' - user with email ' . $email . ' not found inside Insightly');
+            $this->contactRepository->save($uitIdContact);
+            return;
+        }
+
+        $insightlyContact = $insightlyContacts->mostLinks();
+        if (count($insightlyContacts) > 1) {
+            $this->warn($contactId . ' - found multiple contacts with email ' . $email . ' used ' . $insightlyContact->insightlyId);
+        }
+
+        $contact = $this->getContactFromInInsightly($integrationId, $contactId, $insightlyContact->insightlyId);
+        $this->contactRepository->save($contact);
+    }
+
+    private function migrateKeys(UuidInterface $integrationId, MigrationProject $migrationProject): void
+    {
+        if ($migrationProject->apiKeyProduction()) {
+            $consumerProduction = $this->getConsumerFromUitId(
+                $integrationId,
+                $migrationProject->apiKeyProduction(),
+                UiTiDv1Environment::Production
+            );
+            if ($consumerProduction !== null) {
+                $this->uiTiDv1ConsumerRepository->save($consumerProduction);
+            }
+        }
+
+        if ($migrationProject->apiKeyTest()) {
+            $consumerTest = $this->getConsumerFromUitId(
+                $integrationId,
+                $migrationProject->apiKeyTest(),
+                UiTiDv1Environment::Testing
+            );
+            if ($consumerTest !== null) {
+                $this->uiTiDv1ConsumerRepository->save($consumerTest);
+            }
+        }
+    }
+
+    private function getContactFromUiTiD(UuidInterface $integrationId, string $uitId): ?Contact
+    {
+        $response = $this->oauthClientProduction->request(
+            'GET',
+            'user/search?userId=' . $uitId,
+            ['http_errors' => false]
+        );
+        $status = $response->getStatusCode();
+
+        if ($status !== 200) {
+            $this->warn($uitId . ' - did not find user with UiTiD ' . $uitId);
+            return null;
```

Comment:
You could make this less specific,

Preferably something like

`if (!$status->ok)`

I don't know the api

or

`if ($status < 200 || $status > 299)`

Reply:
I wanted to make it very strict. I tested this and only status code `200` is a valid response.

Reply:
Ok fair enough 😉

Reply:
In general I do agree with @simon-debruijn that any 2xx response should be fine in theory, but I also agree with @LucWollants that in this case 200 is the only actual correct response. Because for example `201 Created` or `204 No Content` (other codes that I would accept in other cases) do not make sense for this `GET` request. `200 OK` is the only one that could return the data that we need. (See also https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses)

Reply:
Makes sense indeed 👍🏻